mdal/frmts/mdal_esri_tin.cpp

```diff
@@ -0,0 +1,179 @@
+{
+
+    std::string zFileName = zFile( uri );
+    std::string faceFileName = faceFile( uri );
+
+    std::ifstream xyIn( xyFile( uri ), std::ifstream::in | std::ifstream::binary );
+    if ( ! xyIn.is_open() )
+        return false;
+
+    std::ifstream zIn( zFile( uri ), std::ifstream::in | std::ifstream::binary );
+    if ( ! zIn.is_open() )
+        return false;
+
+    std::ifstream faceIn( faceFile( uri ), std::ifstream::in | std::ifstream::binary );
+    if ( ! faceIn.is_open() )
+        return false;
+
+    //check if the file sizes of the nodes are coherent ?
+
+
+    return true;
+}
+
+std::string MDAL::DriverEsriTin::xyFile(const std::string &uri) const
+{
+    return pathJoin( dirName( uri ), "tnxy.adf" );
+}
+
+std::string MDAL::DriverEsriTin::zFile(const std::string &uri) const
+{
+    return pathJoin( dirName( uri ), "tnz.adf" );
+}
+
+std::string MDAL::DriverEsriTin::faceFile(const std::string &uri) const
+{
+    return pathJoin( dirName( uri ), "tnod.adf" );
+}
+
+void MDAL::DriverEsriTin::populateVertices( const std::string &uri, MDAL::Vertices &vertices )
+{
+    //firt, read the (x,y) coordinates
+    std::ifstream inXY( xyFile( uri ), std::ifstream::in | std::ifstream::binary );
+
+    if ( ! inXY.is_open() )
+        throw MDAL_Status::Err_FileNotFound;
+
+    inXY.ignore( 64 ); //ignore the 64 first bytes
+    //bool readingOk=(!inXY.eof());
+    while ( true )
+    {
+        Vertex vert;
+
+        if ( !readAndChangeEndianness( vert.x, inXY ) )
+            break;
+
+        if ( !readAndChangeEndianness( vert.y, inXY ) )
+            throw MDAL_Status::Err_UnknownFormat;
+
+        vertices.push_back( vert );
+    }
+
+    inXY.close();
+
+    //Then, read Z value
+    std::ifstream inZ( zFile( uri ), std::ifstream::in | std::ifstream::binary );
+  if ( ! inZ.is_open() )
+    throw MDAL_Status::Err_FileNotFound;
+
+  inZ.ignore( 16 ); //ignore the 16 first bytes
+  //readingOk=(!inZ.eof());
+  size_t index = 0;
+  while ( index < vertices.size() )
+  {
+    float zValue;
+    if ( !readAndChangeEndianness( zValue, inZ ) )
+      throw MDAL_Status::Err_UnknownFormat;
+    vertices[index].z = double( zValue );
+    index++;
+
+  }
+
+  inZ.close();
+}
+
+void MDAL::DriverEsriTin::populateFaces( const std::string &uri, MDAL::Faces &faces )
+{
+  std::ifstream inFaces( faceFile( uri ), std::ifstream::in | std::ifstream::binary );
+
+  if ( ! inFaces.is_open() )
+    throw MDAL_Status::Err_FileNotFound;
+
+  while ( true )
+  {
+    Face f;
+    for ( int i = 0; i < 3; ++i )
+    {
+      int ind;
+      if ( !readAndChangeEndianness( ind, inFaces ) )
+        break;
+      f.push_back( static_cast<size_t>( ind ) ); // -5 to ignore "superpoints"
+    }
+
+    if ( f.size() == 0 ) //that's mean this is the end of the file
+      break;
+
+    if ( f.size() < 3 ) //that's mean the face is not complete
+      throw MDAL_Status::Err_UnknownFormat;
+
+    //adjust the indexes and exclude face with "superpoints
```

Comment:
missing "

what is superpoint? can you add description here?

Reply:
Description is in the top of `mdal/frmts/mdal_esri_tin.hpp` file but I will try to be more explicit.

Reply:
thanks